### DISCUSSION TITLE
Can't get REPL on ESP32-C3-13U_kit from AI-Thinker

### DISCUSSION THREAD

**Discussion by myzhan** [micropython/micropython#9923] [Hardware & Peripherals] [answered]
My environment
1. An ESP32-C3-12U_kit from AI-Thinker with 2MB flash.
2. Ubuntu 20.04, ch341-uart converter attached to ttyUSB0.
3. IDF v4.4
4. Connect via "picocom -b 115200 /dev/ttyUSB0"

Tested firmware
1. Prebuilt esp32c3-20220618-v1.19.1.bin from website
2. Prebuilt esp32c3-usb-20220618-v1.19.1.bin from website
3. Compiled by "make BOARD=GENERIC_C3", git hash "eefd946e60aba3ac61c7bfbd0272d07be289e3f3"
4. Compiled by "make BOARD=GENERIC_C3_USB", git hash "eefd946e60aba3ac61c7bfbd0272d07be289e3f3"

And I tried to add these lines to sdkconfig.board, it didn't work.
```
CONFIG_PARTITION_TABLE_CUSTOM=y
CONFIG_PARTITION_TABLE_CUSTOM_FILENAME="partitions-2MiB.csv"
```

See also:
1. https://community.platformio.org/t/board-configuration-for-esp32-c3-2mb-version/22750/2

**Comment by straga** [comment]
You probably need to pull up Enable pin to 3.3v manually: try a piece of wire.

**Reply by straga** [reply]
May be the same #ESP32-C3-01M-Kit
I got an ESP32-C3-01M-Kit  with a CH340 USB Bridge and a big problem: 
If I switch on the serial monitor, the Board stops. 
Using the PCB schematics from HTTP docs.ai-thinker.comThe CH340 RTS output is connected via a 22R resistor to ESP 32 EN and CH340 DTR output is connected via a 22R resistor to ESP 32 GPIO9. 
If you switch on the serial monitor, the CH340 outputs LOW at RTS and DTR and the ESP32-C3 is non-working because of LOW at EN. 
Now I have to press the buttons on the board to start flash programming like with my ESP32 boards, but the serial monitor is working. 

<img width="413" alt="image" src="https://user-images.githubusercontent.com/34532/203305346-dd5a16ac-c315-42b4-a67a-bb3ff870d331.png">

![image](https://user-images.githubusercontent.com/34532/203304827-6929d547-6d99-43fd-846c-28318f757b77.png)
remove resistors.

**Comment by robert-hh** [comment, marked as answer]
There used to be the two transistor logic on ESP boards which implemented a kind of wired XOR, so when both RTS and DTR had the same level, EN and IO0 had a high level, and by making RTS and DTR different levels, either EN or IO0 were low. In an attempt ot save a few cent, vendors started to drop this circuitry. But instead of removing the resistors, one can as well control the level of RTS and DTR at the terminal emulator. For picocom, I use the setting of e.g.:

`picocom --q -b 115200 --noreset --lower-rts --lower-dtr
`

**Reply by myzhan** [reply]
Cool!!! Now I get REPL.
